File: src/main/java/com/rest/app/makePDF/ReportPDF.java
package com.rest.app.makePDF;

import com.rest.app.dataBaseK.Employer;
import com.rest.app.dataBaseK.MyPdfURLs;
import net.sf.jasperreports.engine.*;
import org.jetbrains.annotations.NotNull;

import java.io.File;
import java.util.HashMap;
import java.util.Map;

public class ReportPDF {
    // сотрудники, для вставки данных в отчет
    private final Employer employerEng = new Employer("Ilya Slezkin", "Developer", "8-963-01-65-023", "16.04.1987");
    private final Employer employerRus = new Employer("Илья Слезкин", "Разработчик", "8-963-01-65-023", "16.04.1987");

    public void makeReport(Employer employer, String reportName) throws JRException {
        // DataSource. This is simple example, no database. Then using empty datasource
        JRDataSource dataSource = new JREmptyDataSource();
        // от куда берём jrxml файл
        JasperReport jrxmlFile = JasperCompileManager.compileReport(MyPdfURLs.INSTANCE.getMyReportJrxml());
        // JasperPrint
        JasperPrint jasperPrint = JasperFillManager.fillReport(jrxmlFile, getFillMapParam(employer), dataSource);
        // проверка и создание пути для экспорта PDF файла
        File outDir = new File(MyPdfURLs.INSTANCE.getDirWay());
        outDir.mkdirs();
        // Export to PDF.  "путь/имя экспортируемого PDF файла"
        JasperExportManager.exportReportToPdfFile(jasperPrint, MyPdfURLs.INSTANCE.getExportPDF(reportName));
        //JasperExportManager.exportReportToPdfStream(jasperPrint,);
        System.out.println("method makeReport is done! New file created: " + reportName);
    }

    public void makeTestReport() throws JRException {
        JRDataSource dataSource = new JREmptyDataSource();
        JasperReport jrxmlFile = JasperCompileManager.compileReport(MyPdfURLs.INSTANCE.getMyReportJrxml());
        JasperPrint jasperPrint = JasperFillManager.fillReport(jrxmlFile, getFillMapParam(employerEng), dataSource);
        File outDir = new File(MyPdfURLs.INSTANCE.getDirWay());
        outDir.mkdirs();
        JasperExportManager.exportReportToPdfFile(
                jasperPrint, MyPdfURLs.INSTANCE.getExportPDF("test"));
        System.out.println("method makeTestReport is done! New file created :" + MyPdfURLs.INSTANCE.getExportPDF("test"));
    }

    public void makeRestReport(String userName) throws JRException {
        JRDataSource dataSource = new JREmptyDataSource();
        JasperReport jrxmlFile = JasperCompileManager.compileReport(MyPdfURLs.INSTANCE.getMyReportJrxml());
        JasperPrint jasperPrint = JasperFillManager.fillReport(jrxmlFile, getFillMapParam(employerEng), dataSource);
        File outDir = new File(MyPdfURLs.INSTANCE.getDirWay());
        outDir.mkdirs();
        JasperExportManager.exportReportToPdfFile(jasperPrint, MyPdfURLs.INSTANCE.getExportPDF(userName));
        System.out.println("method makeTestReport is done! New file created :" + MyPdfURLs.INSTANCE.getExportPDF(userName));
    }

    @NotNull
    // метод для заполнения Мапы и получения её parameters
    protected Map<String, Object> getFillMapParam(Employer employer) {
        Map<String, Object> parameters = new HashMap<>(); // Parameters for report
        // Parameters for report
        parameters.put("jr_name", employer.getJrName());
        parameters.put("jr_position", employer.getJrPosition());
        parameters.put("jr_phone_mobile", employer.getJrPhoneMobile());
        parameters.put("jr_data_birthday", employer.getJrDataBirthday());
        return parameters;
    }

    public Employer getEmployerEng() {
        return employerEng;
    }

    public Employer getEmployerRus() {
        return employerRus;
    }
}

File: src/main/java/com/rest/app/makePDF/ReportPDF.kt
package com.rest.app.makePDF

import com.rest.app.dataBase.Employer
import com.rest.app.dataBase.EmployerGet
import com.rest.app.dataBase.GetMap
import com.rest.app.dataBase.MyPdfURLs.getDirWay
import com.rest.app.dataBase.MyPdfURLs.getExportPDF
import com.rest.app.dataBase.MyPdfURLs.getMyReportJrxml
import net.sf.jasperreports.engine.*
import java.io.File

class ReportPDF {
    private val employerGet = EmployerGet()     // ссылка на класс с сотрудниками
    private val getMap = GetMap()   // ссылка на мапу

    // генерирует PDF на сервере
    @Throws(JRException::class)
    fun makeReport(employer: Employer, reportName: String) {
        // DataSource. This is simple example, no database. Then using empty datasource
        val dataSource: JRDataSource = JREmptyDataSource() // обязательно использовать! без него будут пустые отчеты
        // от куда берём jrxml файл
        val jrxmlFile = JasperCompileManager.compileReport(getMyReportJrxml())
        // JasperPrint - заполняем шаблон
        val jasperPrint = JasperFillManager.fillReport(jrxmlFile, getMap.getFillMap(employer), dataSource)
        // проверка и создание пути для экспорта PDF файла
        val outDir = File(getDirWay())
        outDir.mkdirs()
        // Экспорт данных в PDF файл
        JasperExportManager.exportReportToPdfFile(jasperPrint, getExportPDF(reportName))    // ,(заполненный шаблон, путь(имя))
        // для отчёта
        println("method makeReport is done! New file created: $reportName")
    }

    // для RESTа - генерирует PDF на сервере
    @Throws(JRException::class)
    fun makeRestReport(userName: String) {
        val dataSource: JRDataSource = JREmptyDataSource()
        val jrxmlFile = JasperCompileManager.compileReport(getMyReportJrxml())
        val jasperPrint = JasperFillManager.fillReport(jrxmlFile, getMap.getFillMap(employerGet.getEnglish()), dataSource)
        val outDir = File(getDirWay())
        outDir.mkdirs()
        JasperExportManager.exportReportToPdfFile(jasperPrint, getExportPDF(userName))
        println("method makeTestReport is done! New file created :" + getExportPDF(userName))
    }
}
